File: core/src/main/java/co/ericp/freewayforecast/weather/WeatherSource.java
package co.ericp.freewayforecast.weather;

import co.ericp.freewayforecast.LatLng;
import co.ericp.freewayforecast.weather.Forecast;
import rx.Single;

/**
 * A source for getting the weather.
 *
 * An example implementation of this would be the weather.com API.
 */
public interface WeatherSource {
    Single<Forecast> getForecast(LatLng coords, long time);
}

File: core/src/main/java/co/ericp/freewayforecast/weather/WeatherSource.kt
package co.ericp.freewayforecast.weather

import co.ericp.freewayforecast.Location
import io.reactivex.Observable

/**
 * A source for getting the weather.

 * An example implementation of this would be the weather.com API.
 */
interface WeatherSource {
    fun getForecast(location: Location,
                    time: Long,
                    until: Long? = null): Observable<WeatherPoint>
}
